python/lsst/cp/pipe/measureCrosstalk.py

```diff
@@ -0,0 +1,799 @@
+            plot.title(f"(Source {i} -> Target {j}) mean: {value:.2g} coeff: {coeff:.2g} valid: {valid}")
+            figure.show()
+
+            prompt = "Press Enter to continue: "
+            while True:
+                ans = input(prompt).lower()
+                if ans in ("", "c",):
+                    break
+                elif ans in ("pdb", "p",):
+                    import pdb
+                    pdb.set_trace()
+            plot.close()
+
+
+class MeasureCrosstalkConfig(Config):
+    extract = ConfigurableField(
+        target=CrosstalkExtractTask,
+        doc="Task to measure pixel ratios.",
+    )
+    solver = ConfigurableField(
+        target=CrosstalkSolveTask,
+        doc="Task to convert ratio lists to crosstalk coefficients.",
+    )
+
+
+class MeasureCrosstalkTask(pipeBase.CmdLineTask):
+    """Measure intra-detector crosstalk.
+
+    See also
+    --------
+    lsst.ip.isr.crosstalk.CrosstalkCalib
+    lsst.cp.pipe.measureCrosstalk.CrosstalkExtractTask
+    lsst.cp.pipe.measureCrosstalk.CrosstalkSolveTask
+
+    Notes
+    -----
+    The crosstalk this method measures assumes that when a bright
+    pixel is found in one detector amplifier, all other detector
+    amplifiers may see an increase in the same pixel location
+    (relative to the readout amplifier) as these other pixels are read
+    out at the same time.
+
+    After processing each input exposure through a limited set of ISR
+    stages, bright unmasked pixels above the threshold are identified.
+    The potential CT signal is found by taking the ratio of the
+    appropriate background-subtracted pixel value on the other
+    amplifiers to the input value on the source amplifier.  If the
+    source amplifier has a large number of bright pixels as well, the
+    background level may be elevated, leading to poor ratio
+    measurements.
+
+    The set of ratios found between each pair of amplifiers across all
+    input exposures is then gathered to produce the final CT
+    coefficients.  The sigma-clipped mean and sigma are returned from
+    these sets of ratios, with the coefficient to supply to the ISR
+    CrosstalkTask() being the multiplicative inverse of these values.
+
+    This Task simply calls the pipetask versions of the measure
+    crosstalk code.
+    """
+    ConfigClass = MeasureCrosstalkConfig
+    _DefaultName = "measureCrosstalk"
+
+    # Let's use this instead of messing with parseAndRun.
+    RunnerClass = DataRefListRunner
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("extract")
+        self.makeSubtask("solver")
+
+    def runDataRef(self, dataRefList):
+        """Run extract task on each of inputs in the dataRef list, then pass
+        that to the solver task.
+
+        Parameters
+        ----------
+        dataRefList : `list` [`lsst.daf.peristence.ButlerDataRef`]
+            Data references for detectors to process.
```

Comment:
Gen2 only solves intra-detector, and because it can't access the dimension information generated by the Gen3 pipetask, the output detector needs to be set  manually.  There's a check on line line 457 that raises if multiple detectors are supplied, but as that's in the solve task, I've added another raise in the gen2 runDataRef.

Reply:
Perfect.